internal/appsec/remoteconfig.go

```diff
@@ -228,17 +332,35 @@ func (a *appsec) enableRemoteActivation() error {
 	}
 	a.registerRCProduct(rc.ProductASMFeatures)
 	a.registerRCCapability(remoteconfig.ASMActivation)
-	a.registerRCCallback(a.asmFeaturesCallback, rc.ProductASMFeatures)
 	return nil
 }
 
-func (a *appsec) enableRCBlocking(handle wafHandleWrapper) error {
+func (a *appsec) enableRCBlocking() {
 	if a.rc == nil {
-		return fmt.Errorf("no valid remote configuration client")
+		log.Debug("appsec: Remote config: no valid remote configuration client")
+		return
 	}
+
+	a.registerRCProduct(rc.ProductASM)
+	a.registerRCProduct(rc.ProductASMDD)
 	a.registerRCProduct(rc.ProductASMData)
-	a.registerRCCapability(remoteconfig.ASMIPBlocking)
 	a.registerRCCapability(remoteconfig.ASMUserBlocking)
-	a.registerRCCallback(handle.asmDataCallback, rc.ProductASMData)
-	return nil
+	a.registerRCCapability(remoteconfig.ASMRequestBlocking)
+
+	if _, isSet := os.LookupEnv(rulesEnvVar); !isSet {
+		a.registerRCCapability(remoteconfig.ASMIPBlocking)
+		a.registerRCCapability(remoteconfig.ASMDDRules)
+		a.registerRCCapability(remoteconfig.ASMExclusions)
+	}
+}
+
+func (a *appsec) disableRCBlocking() {
+	if a.rc == nil {
+		return
+	}
+	a.unregisterRCCapability(remoteconfig.ASMDDRules)
+	a.unregisterRCCapability(remoteconfig.ASMExclusions)
+	a.unregisterRCCapability(remoteconfig.ASMIPBlocking)
+	a.unregisterRCCapability(remoteconfig.ASMRequestBlocking)
+	a.unregisterRCCapability(remoteconfig.ASMUserBlocking)
```

Comment:
do we need to unregister the product subscriptions too?

Reply:
Summary of our discussion here: no need